ansible/bootstrap.yml

```diff
@@ -22,14 +22,52 @@
         state: "{{ selinux_state }}"
         policy: "{{ selinux_policy }}"
       register: sestatus
-    - name: Reboot if required to change SELinux state
+
+- hosts: update
+  gather_facts: false
+  become: yes
+  tags:
+    - update
+  tasks:
+    - block:
+      - name: Update selected packages
+        yum:
+          name: "{{ update_name }}"
+          state: "{{ update_state }}"
+          exclude: "{{ update_exclude }}"
+          disablerepo: "{{ update_disablerepo }}"
+        register: updates
+      - debug:
+          var: updates
+      - name: Log updated packages
+        copy:
+          content: "{{ updates.results | join('\n') }}"
+          dest: "{{ update_log_path }}"
+        delegate_to: localhost
+      - debug:
+          msg: "{{ updates.results | length }} changes to packages - see {{ update_log_path }} for details"
+      when: "update_enable | default('false') | bool"
+
+- hosts:
+    - selinux
+    - update
+  gather_facts: false
+  become: yes
+  tags:
+    - reboot
+  tasks:
+    - name: Check for pending reboot from package updates
+      stat: 
+        path: /var/run/reboot-required
+      register: update_reboot_required
+    - debug:
+        msg: "setstatus:{{ (sestatus.reboot_required | default(false)) }} packages: {{ (update_reboot_required.stat.exists | bool) }}"
+    - name: Reboot if required from SELinux state change or package upgrades
```

Comment:
Nice work, mostly looks good to me. This is more of a thought than a suggestion, so feel free to shoot it down. Do you reckon this could be simplified if you use a handler? Could potentially notify on  `/var/run/reboot-required` existing or sestatus.reboot_required? Might need some cajoling on changed_when or a separate task.

Reply:
No as not inside a role.